src/hotspot/share/prims/jvmtiThreadState.hpp

```diff
@@ -83,6 +83,8 @@ class JvmtiVTMSTransitionDisabler {
   static volatile int _VTMS_transition_disable_for_all_count; // transitions for all virtual threads are disabled while it is positive
   static volatile bool _SR_mode;                         // there is an active suspender or resumer
   static volatile int _VTMS_transition_count;            // current number of VTMS transitions
+  static int  _sync_protocol_enabled_count;              // current number of JvmtiVTMSTransitionDisablers enabled sync protocol
```

Comment:
The _sync_protocol_enabled_count and  _sync_protocol_enabled_permanently are read/updated in different threads. How access to them is protected from racing? Might be make sense to add this info in comment?

Reply:
Good catch, thanks. My initial intention was to make them volatile with Atomic load/store/update. Fixed now.